### DISCUSSION TITLE
Introduction to Makie and conceptual basis

### DISCUSSION THREAD

**Discussion by asinghvi17** [JuliaPlots/Makie.jl#2854] [Ideas]
This is feedback I've heard from multiple people - coming into the Makie ecosystem with no prior experience is a whirlwind of new concepts with nothing conceptually tying them together.  

We should have an intro page which covers the basic theoretical foundations of Makie:
- Observables, all attributes and inputs are observables, how you can play around with these
- What are figures, scenes, and plots?  How do they relate to each other?
- Why do some plots have plots inside of them?  How does Makie actually draw?
- What is a layout?
- What is the Scene/Plot graph?
- Best practices for Makie (don't keep making new figures/plots, instead update existing attributes and data)

and most importantly, *why should you use Makie*?

I know these are covered in tutorials.  The issue is that the tutorials also cover a lot of other stuff.  This would be a short document with hardly any plots, but more extremely simple and to-the-point code and a lot of writing.

What do people here think?

**Comment by behinger** [comment]
This is great!

> What are figures, scenes, and plots? How do they relate to each other?

And Axes / FigureGrids / Figure.layouts etc - I am so confused :S

**Comment by igaszczesniak** [comment]
Great idea! This is exactly what I'm looking for :)

**Comment by timholy** [comment]
https://documentation.divio.com/, lower-left quadrant. If you've never seen this before, it's worth working your way through the whole site. I regret not knowing about this when I wrote the JuliaImages and SnoopCompile docs.

**Reply by asinghvi17** [reply]
This is really neat, thanks!  Will have to give it a thorough read.  I agree that this explanation falls in the lower left category.  There was an impromptu seminar at MakieCon about these foundational concepts where Julius and I spoke, and I will probably write that whole conversation down as a nice explanation about the theoretical underpinnings of Makie.   Xref #2866

**Comment by asinghvi17** [comment]
Notes from Bogumil's talk

- Be more opinionated on backends on the landing page
- You have to think in layers/layering...plot! just adds another layer.
- More description about which attributes map to which types
- Emphasis on integration across the ecosystem

**Comment by asinghvi17** [comment]
A few thoughts on the plot/recipe interface:

1. Where can you "buy in" to the ecosystem? (`convert_arguments` or `plot!`)
2. What are the advantages of one over the other in these points?  Converts are simple and broadly applicable, whereas plot! is for fine control when you know what you want to do.
3. How can you dictate how a certain recipe is rendered on a certain backend?  (Refer to e.g. `poly` in CairoMakie)
4. What endpoints of the interface do you have to implement for your recipe?  (`plottype(args...)`, `used_attributes`, etc)
5. What other optional endpoints exist?  (`data_limits`, etc)
